widget-showcase/tests/unit/app.ts

```diff
@@ -0,0 +1,789 @@
+
+		widget.expectRender(expectedWidget, 'The select box is changed');
+	},
+
+	'can change the multi select'() {
+		widget.callListener('onChange', {
+			key: 'multiselect',
+			args: [{
+				value: 'New option value'
+			}]
+		});
+
+		expected(widget);
+		const expectedWidget = expected(widget);
+
+		assignProperties(findKey(expectedWidget, 'multiselect')!, {
+			value: 'New option value'
+		});
+
+		widget.expectRender(expectedWidget, 'The multi select box is changed');
+	},
+
+	'can change the slider value'() {
+		widget.callListener('onInput', {
+			key: 'slider',
+			args: [{
+				target: {
+					value: '2'
+				}
+			}]
+		});
+
+		expected(widget);
+		const expectedWidget = expected(widget);
+
+		/*
+			A workaround - https://github.com/dojo/test-extras/blob/42a94c2b711d6d8abece5ccf54e8ddb2f0cc2993/src/support/d.ts#L20 - VirtualDomProperties have an interface of: readonly value?: string;
+		*/
+
+		assignProperties(findKey(expectedWidget, 'slider')!, {
+			value: 2
+		} as any);
+
+		widget.expectRender(expectedWidget, 'The slider value is set');
+	},
+
+	'can change the text area value'() {
+		widget.callListener('onChange', {
+			key: 'text-area',
+			args: [{
+				target: {
+					value: 'some textarea value'
+				}
+			}]
+		});
+
+		expected(widget);
+		const expectedWidget = expected(widget);
+
+		assignProperties(findKey(expectedWidget, 'text-area')!, {
+			value: 'some textarea value'
+		});
+
+		widget.expectRender(expectedWidget, 'The textarea value is set');
+	},
+
+	'can open and close a dialog'() {
+		widget.callListener('onClick', {
+			key: 'dialog-button'
+		});
+
+		expected(widget);
+		const expectedWidget = expected(widget);
+
+		assignProperties(findKey(expectedWidget, 'dialog')!, {
+			open: true
+		});
+
+		widget.expectRender(expectedWidget, 'the dialog opens');
+
+		widget.callListener('onRequestClose', {
+			key: 'dialog'
+		});
+
+		assignProperties(findKey(expectedWidget, 'dialog')!, {
+			open: false
+		});
+
+		widget.expectRender(expectedWidget, 'the dialog closes');
+	},
+
+	'can open and close a slide pane'() {
+		widget.callListener('onClick', {
+			key: 'slidepane-button'
+		});
+
+		expected(widget);
+		const expectedWidget = expected(widget);
+
+		assignProperties(findKey(expectedWidget, 'slide-pane')!, {
+			open: true
+		});
+
+		widget.expectRender(expectedWidget, 'the slide pane opens');
+
+		widget.callListener('onRequestClose', {
+			key: 'slide-pane'
+		});
+
+		assignProperties(findKey(expectedWidget, 'slide-pane')!, {
+			open: false
+		});
+
+		widget.expectRender(expectedWidget, 'the slide pane closes');
+	},
+
+	'can resize a split pane'() {
+		widget.callListener('onResize', {
+			key: 'split-pane',
+			args: [5]
+		});
+
+		expected(widget);
+		const expectedWidget = expected(widget);
+
+		assignProperties(findKey(expectedWidget, 'split-pane')!, {
+			size: 5
+		});
+
+		widget.expectRender(expectedWidget, 'The split pane size is set');
+
+		/*
+
+		TODO: How to test a nested split pane?
```

Comment:
@agubler could you help with this TODO please? How do you think I should test this? https://github.com/dojo/examples/blob/0291f47482fb24616bd0d229eef313a9ec787a24/widget-showcase/src/App.ts#L356-L364